docs/query_proxy.md

```diff
@@ -0,0 +1,205 @@
+REST endpoint for Wormhole query requests. It validates those requests and forwards them to the guardian
+CCQ P2P network for processing by the guardians. It then accumulates the responses from the guardians,
+verifies quorum and forwards the response to the client.
+
+## Building the Proxy Server
+
+The proxy server runs as another instance of the `guardiand` process, similar to the spy. It is built exactly
+the same as the spy, and requires the same dependencies. Please see the [Operations Guide](operations.md) for
+details on how to build `guardiand`.
+
+## Deploying the Proxy Server
+
+The proxy server can be deployed just like the spy, including potentially running in a container. Note that it
+requires a public IP address to listen for REST requests, and it needs to be able to reach the guardian P2P network.
+
+The proxy is not particularly resource intensive, so should run successfully on a reasonable size VM.
+
+## Configuring the Proxy Server
+
+There are two main parts to configuring the proxy server. The first is setting up the command line arguments,
+which generally will not change after initial setup. The second part of the configuration is the permissions file,
+which will change as the requirements of integrators change.
+
+### Proxy Server Command Line Arguments
+
+The following is a sample command line for running the proxy server in mainnet.
+
+```code
+
+build/bin/guardiand query-server \
+      --env "mainnet" \
+      --nodeKey /home/ccq/data/ccq_server.nodeKey \
+      --permFile "/home/ccq/data/ccq_server.perms.json" \
+      --signerKey "/home/ccq/data/ccq_server.signerKey" \
+      --listenAddr "[::]:80" \
+      --ethRPC https://eth.drpc.org \
+      --ethContract "0x98f3c9e6E3fAce36bAAd05FE09d375Ef1464288B" \
+      --logLevel=info \
+      --telemetryLokiURL $LOKI_URL \
+      --telemetryNodeName "Mainnet CCQ server 1" \
+      --promRemoteURL $PROM_URL
+
+```
+
+- The `env` can be mainnet, testnet or devnet.
+- The `nodeKey` should point to the file containing the P2P key. The first time the proxy runs, if the
+  file does not exist, it will be created. You can look in the proxy server logs to get the generated key.
+- The `permFile` is the JSON permissions file, which is documented below.
+- The `signerKey` should point to an armored file containing a key that will be used to sign requests received
+  from integrators who are configured to support auto signing and opt not to sign a request. Please see below
+  for how to generate this file.
+- The `listenAddr` specifies the port on which the proxy listens for REST requests.
+- The `ethRPC` and `ethContract` are used to read the wormhole guardian set on start up. The address
+  above is for mainnet. If you are running in testnet, you should point to Holesky and use `0xa10f2eF61dE1f19f586ab8B6F2EbA89bACE63F7a`.
+  Note that using a public endpoint should be fine, since the proxy only does a single read of the guardian set.
```

Comment:
Consider linking to the [Contract Addresses](https://docs.wormhole.com/wormhole/reference/constants#contract-addresses) table in the official docs.